### PR TITLE
adding basic tests

### LINKED ISSUE #2008: make tests for synonym server
### Issue

- [ ] This is a bug report.
- [x] This is a feature request.

### Description

[Description of the bug or feature.]
[Include images where appropriate]

### Links to feature design doc, conversations, mockups, etc.
I needed to add tests for server functions so I could develop in a more regular way.

### Stakeholders
myself and anyone that likes having tests.

### Functional Requirements:
Tests existing synonym server functions

### Acceptance Criteria:
tests exist for each function in the synonyms server.

### PR DESCRIPTION
**Pull request by Courey**
### Title   
 feat(testing): add testing for existing synonym server functions

### Description
Just adding some tests. Largely mocks, but testing what I can.

### Reviewers
@lpsinger - he's familiar with the work

### Changes
adding tests

### Acceptance Criteria
this was not a request that had acceptance criteria. I just needed to do it because I can't develop well with no tests and no UI.

### Related Issue(s)
Closes #2008 

### Testing
npm run tests - all tests are passing

### CODE REVIEW

**Comment by Courey** [comment]
there are more changes to the package-lock.json than I would have expected. Looking into it.

**Comment by lpsinger** [comment]
@Courey, this is very timely as it is related to the testing approach in #2009.

**Comment by Courey** [comment]
This is fine to go in once reviewed. I rebased it so it's fresh since it was a little stale.

File: app/routes/_gcn.synonyms/synonyms.server.ts [hunk 1]
```diff
@@ -110,7 +110,7 @@ export async function searchSynonymsByEventId({
  * BatchWriteItem has a limit of 25 items, so the user may not add more than
  * 25 synonyms at a time.
  */
-export async function createSynonyms(...synonymousEventIds: string[]) {
+export async function createSynonyms(synonymousEventIds: string[]) {
```
Comment (lpsinger) [inline app/routes/_gcn.synonyms/synonyms.server.ts:113]: Why did you change the arguments back?

**Review by lpsinger — CHANGES_REQUESTED** [review]

File: app/routes/_gcn.synonyms/synonyms.server.ts (hunk 1 above)
Comment (Courey) [reply app/routes/_gcn.synonyms/synonyms.server.ts:113]: becuse I want to pass in an array of string eventIds [like this](https://github.com/nasa-gcn/gcn.nasa.gov/pull/2007/files#diff-0c3c91f87196d09d921467bd436eb5cfc88043d719ed1ffef34ea277148c875cR50)
```
const synonymousEventIds = ['eventId1', 'eventId2']
const result = await createSynonyms(synonymousEventIds)
```
I do NOT want to pass in eventIds like this:
```
createSynonyms('eventId1', 'eventId2', 'eventId3')
```
unless there is a reason for doing it that way

Comment (lpsinger) [reply app/routes/_gcn.synonyms/synonyms.server.ts:113]: Well, you can call it like this:

```js
createSynonyms(...synonymousEventIds)
```

It doesn't really matter, as long as it's consistent across similar functions in the backend.
